src/lib/work_partitioner/pending_zkapp_command.ml

```diff
@@ -0,0 +1,48 @@
+(* NOTE: this module is where the real optimization happen. One assumption we
+   have is the order of merging is irrelvant to the correctness of the final
+   proof. Hence we're only using a counter `merge_remaining` to track have we
+   reach the final proof
+*)
+
+open Core_kernel
+open Snark_work_lib
+
+type t =
+  { job : (Spec.Single.t, Id.Single.t) With_job_meta.t
+        (* the original work being splitted, should be identical to Work_selector.work *)
+  ; unscheduled_segments : Spec.Sub_zkapp.t Queue.t
+  ; pending_mergable_proofs : Ledger_proof.Cached.t Deque.t
```

Comment:
```suggestion
  ; pending_mergeable_proofs : Ledger_proof.Cached.t Deque.t
```

Reply:
Also, probably we don't need to cache these short-lived proofs, it's best to keep them `Ledger_proof.t` without conversion to `Ledger_proof.Cached.t`.